h2o-automl/src/main/java/ai/h2o/automl/preprocessing/TargetEncoding.java

```diff
@@ -0,0 +1,126 @@
+package ai.h2o.automl.preprocessing;
+
+import ai.h2o.automl.AutoML;
+import ai.h2o.targetencoding.TargetEncoder;
+import ai.h2o.targetencoding.TargetEncoderModel;
+import ai.h2o.targetencoding.TargetEncoderModel.DataLeakageHandlingStrategy;
+import ai.h2o.targetencoding.TargetEncoderModel.TargetEncoderParameters;
+import ai.h2o.targetencoding.TargetEncoderPreprocessor;
+import hex.Model;
+import hex.Model.Parameters.FoldAssignmentScheme;
+import hex.ModelPreprocessor;
+import water.DKV;
+import water.Key;
+import water.fvec.Frame;
+import water.fvec.Vec;
+import water.rapids.ast.prims.advmath.AstKFold;
+import water.util.ArrayUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TargetEncoding implements PreprocessingStep {
+    
+    private transient AutoML _aml;
+    private transient TargetEncoderPreprocessor _tePreprocessor;
+    private transient TargetEncoderModel _teModel;
+    private final transient List<Completer> _disposables = new ArrayList<>();
+
+    public TargetEncoding(AutoML aml) {
+        _aml = aml;
+    }
+
+    @Override
+    public void prepare() {
+        TargetEncoderParameters params = new TargetEncoderParameters();
+        params._train = _aml.getTrainingFrame()._key;
+        params._response_column = _aml.getBuildSpec().input_spec.response_column;
+        params._keep_original_features = false;
+        params._blending = true;
+        params._noise = 0;
```

Comment:
differ from what I wrote in the ticket: 
- decided to use default blending params instead so that it behaves better by default with categories with "rare" occurences.
- and disabled noise as it is considered difficult to tune correctly: we can't expect the "auto" noise to be good in most cases.